File: src/talker_listener_py/talker_listener_py/listener.py
import rclpy
from rclpy.node import Node

from std_msgs.msg import String

class Listener(Node):

    def __init__(self):
        super().__init__("listener_py")

        self.subscriber = self.create_subscription(
            String,
            "talker",
            self.listener_callback,
            10
            )

    def listener_callback(self, msg: String):
        
        self.get_logger().info(f"Listener received messaged: {msg.data}")


def main(args=None):
    rclpy.init(args=args)

    listener = Listener()
    rclpy.spin(listener)

if __name__ == "__main__":
    main()

File: src/talker_listener_cpp/src/listener.cpp
#include <rclcpp/rclcpp.hpp>
#include <std_msgs/msg/string.hpp>


class Listener : public rclcpp::Node {
    public:
        Listener() : Node("listener_cpp") {
            // Create subscription
            subscription_ = this->create_subscription<std_msgs::msg::String>(
                "talker", 10, std::bind(&Listener::listener_callback, this, std::placeholders::_1)
            );
        }
    private:
        void listener_callback(const std_msgs::msg::String::SharedPtr msg) {
            RCLCPP_INFO(this->get_logger(), "Received: %s", msg->data.c_str());
        }

    // Member variables
    rclcpp::Subscription<std_msgs::msg::String>::SharedPtr subscription_;   
};


int main(int argc, char *argv[]) {
  rclcpp::init(argc, argv);
  rclcpp::spin(std::make_shared<Listener>());
  
  return 0;
}
